### PR TITLE
D8CORE-3743 Use limited cookie to temporarily save form data

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Use a 5 second cookie to temporarily store form data in case of form validation issues.
   - The cookie is unique to the specific field
   - The cookie is only valid on the same url path so it won't exist on the actual node page or simultaneous node forms.
   - 5 seconds should be enough to load the react javascript and then the cookie goes away.

# Need Review By (Date)
- 4/21 (unless we get it in this week, we can push this out as a hotfix)

# Urgency
- medium

# Steps to Test
1. checkout this branch & clear browser caches
2. create a publication and choose the citation type "Article Newspaper/Magazine"
3. enter "01" into the month or day field (be sure it has a zero as the preceding character)
4. now populate some paragraph components with some values.
5. save and see the validation produces an error on the form
6. verify your paragraphs you populated are still there and configured correctly.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

**Comment by sherakama** [comment]
What problem are you trying to solve here?

**Comment by pookmish** [comment]
@sherakama what we had before worked if the error occurred on the main form. But it didn't work when there was an error on an inline entity form. The error occurs outside of the primary form and so the form isn't rebuilt with the changes from the react ares. This solves those issues

File: js/src/Contexts/WidgetManager.js [hunk 1]
```diff
@@ -102,6 +95,25 @@ export class WidgetManager extends Component {
...
   }
 
+  getCookie() {
+    const formItemsField = document.getElementById(this.props.inputId);
+    const cname = formItemsField.getAttribute('data-cookie-id');
+
+    var name = cname + "=";
+    var decodedCookie = decodeURIComponent(document.cookie);
+    var ca = decodedCookie.split(';');
+    for(var i = 0; i <ca.length; i++) {
```
Comment (imonroe) [inline js/src/Contexts/WidgetManager.js]:
```suggestion
    for(var i = 0; i < ca.length; i++) {
```

File: js/src/Contexts/WidgetManager.js [hunk 2]
```diff
@@ -102,6 +95,25 @@ export class WidgetManager extends Component {
     this.componentDidUpdate();
   }
 
+  getCookie() {
```
Comment (imonroe) [inline js/src/Contexts/WidgetManager.js:98]: This seems like a really hard way to do this.  It looks like you're decoding the cookie by parsing it character by character.  Surely there must be some cookie library or something that we could use that would be a more efficient solution?

**Review by imonroe — APPROVED** [review]
The functionality works as described.  I'll approve it, but I think we may want to consider a more robust solution to managing the cookies.  Seems like we're doing it the hard way.

File: js/src/Contexts/WidgetManager.js (hunk 2 above)
Comment (pookmish) [reply js/src/Contexts/WidgetManager.js:98]: well there's this https://www.npmjs.com/package/react-cookie but that adds some unnecessary weight for this task.

**Pull request merged**